### PR TITLE
This is a related change to the fix of issue NuGet/Home#1158. This change fixes auth in the command line.

### PR DESCRIPTION
**Pull request by feiling**
The changes here are basically the same change in NuGet.VisualStudio

@deepakaravindr @emgarten @zhili1208 @yishaigalatzer

### CODE REVIEW

File: src/NuGet.CommandLine/Commands/Command.cs [hunk 1]
```diff
@@ -47,6 +49,8 @@ protected Command()
         [Option(typeof(NuGetCommand), "Option_ConfigFile")]
         public string ConfigFile { get; set; }
 
+        private HashSet<Uri> _credentialRequested;
```
Comment (emgarten) [inline src/NuGet.CommandLine/Commands/Command.cs]: this can be readonly

File: src/NuGet.CommandLine/Commands/Command.cs [hunk 2]
```diff
@@ -152,6 +156,36 @@ protected void SetDefaultCredentialProvider()
...
+                }
+
+                return credentialProvider.GetCredentials(
+                    uri,
+                    proxy: null,
+                    credentialType: CredentialType.RequestCredentials,
+                    retrying: retrying);
+            };
+
+            var v2CredentialStoreType = typeof(NuGet.ICredentialCache).Assembly.GetType("NuGet.CredentialStore");
```
Comment (emgarten) [inline src/NuGet.CommandLine/Commands/Command.cs]: We should not be doing reflection for this. Can you just make it public in NuGet.Core?

**Comment by emgarten** [comment]
Remove the reflection :watch:

File: src/NuGet.CommandLine/Commands/Command.cs (hunk 1 above)
Comment (feiling) [inline src/NuGet.CommandLine/Commands/Command.cs]: Good point. fixed.

File: src/NuGet.CommandLine/Commands/Command.cs (hunk 2 above)
Comment (feiling) [inline src/NuGet.CommandLine/Commands/Command.cs]: Reflection removed.

**Pull request closed without merging**